.github/workflows/preview-release-on-comment.yml

```diff
@@ -0,0 +1,99 @@
+name: Preview Release on Comment
+
+permissions:
+  contents: read
+  pull-requests: write
+  issues: write
+  statuses: write
+
+on:
+  issue_comment:
+    types: [created]
+
+jobs:
+  release:
+    if: github.event.issue.pull_request && startsWith(github.event.comment.body, '[preview_deployment]')
+    runs-on: ubuntu-latest
+
+    timeout-minutes: 15
+
+    steps:
+      # Get GitHub token via the CT Changesets App
+      - name: Generate GitHub token (via CT Changesets App)
+        id: generate_github_token
+        uses: tibdex/github-app-token@v2.1.0
+        with:
+          app_id: ${{ secrets.CT_CHANGESETS_APP_ID }}
+          private_key: ${{ secrets.CT_CHANGESETS_APP_PEM }}
+      
+      - name: Get branch of PR
+        uses: xt0rted/pull-request-comment-branch@v1
+        id: comment-branch
+      
+      - name: Ensure branch name matches 'preview/**'
+        run: |
+          BRANCH_NAME=${{ steps.comment-branch.outputs.head_ref }}
+          if [[ ! "$BRANCH_NAME" =~ ^preview/ ]]; then
+            echo "Branch name does not match 'preview/**'. Exiting workflow."
+            exit 1
+          fi
```

Comment:
Thanks Rotimi for working on this 🙌
Just curious – is there a reason we’re only allowing this on branches starting with `preview/*`? It could make things a bit trickier, like for PRs from Renovate, since we'd have to duplicate those branches and PRs.

Reply:
This was a decision to avoid the release being triggered from any branch. cc @CarlosCortizasCT